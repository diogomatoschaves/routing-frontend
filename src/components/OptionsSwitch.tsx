import React from 'react'
import styled from 'styled-components'
import { Box } from '../styledComponents'
import { Checkbox } from 'semantic-ui-react'
import { UpdateState } from '../types'


const StyledText = styled.label`
  width: 75%;
  margin: 0;
  padding-left: 15px;
  font-size: 16px;
  color: rgb(70, 70, 70);
  font-weight: 200;
`
const StyledCheckbox = styled(Checkbox)`
  width: 25%;
  &.ui.checked.fitted.toggle.checkbox input:checked~label:before {
    background-color: #79aebf !important;
  }
`

interface Props {
  checked: boolean,
  text: string,
  id: string,
  updateState: UpdateState,
  width?: string
}


const OptionsSwitch = ({ checked, text, id, updateState, width }: Props) => {
  return (
    <Box direction="row" height="50px" width={width ? width : undefined}>
      <StyledCheckbox 
        id={id}
        className="custom-class"
        checked={checked} 
        onChange={(e: any, { checked }: { checked: boolean }) => updateState(id, checked)}
        toggle
      />
      <StyledText>{text}</StyledText>
    </Box>
  )
}

export default OptionsSwitch
