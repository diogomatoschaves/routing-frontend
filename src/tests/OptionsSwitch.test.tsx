import React from 'react'
import TestRenderer from 'react-test-renderer'
import OptionsSwitch from '../components/OptionsSwitch'
import { MemoryRouter } from 'react-router-dom'

it('expect to render OptionsPanel component', () => {

  const testInstance = TestRenderer.create(
    <MemoryRouter initialEntries={[ '/' ]}>
      <OptionsSwitch
        checked={false}
        text={'Google Maps'}
        id={'googleMapsOption'}
        updateState={jest.fn()}
        width={'65%'}
      />
    </MemoryRouter>
  )
  expect(testInstance.toJSON()).toMatchSnapshot()
}) 
