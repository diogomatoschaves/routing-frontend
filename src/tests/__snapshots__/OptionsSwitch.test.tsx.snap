// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render OptionsPanel component 1`] = `
<div
  className="sc-bdVaJa dbyCSG"
  direction="row"
  height="50px"
  width="65%"
>
  <div
    className="ui fitted toggle checkbox custom-class sc-gqjmRU kbDkMN"
    onChange={[Function]}
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
  >
    <input
      checked={false}
      className="hidden"
      id="googleMapsOption"
      readOnly={true}
      tabIndex={0}
      type="checkbox"
    />
    <label
      htmlFor="googleMapsOption"
    />
  </div>
  <label
    className="sc-gZMcBi eCHFCo"
  >
    Google Maps
  </label>
</div>
`;
